Installing-and-Configuring/RecoveryShell.md

```diff
@@ -70,7 +70,7 @@ This example may work for you by changing only the root= setting.  Normally, the
 
 ### sign files
 
-All files in /boot are signed using the security dongle paired with heads.  These signatures are stored in /boot/kexec_hashes.txt
+All files in /boot are signed using the security dongle paired with heads.  These signatures are stored in /boot/kexec.sig
```

Comment:
Unfortunately it is more complicated then that.

kexec.sig is a detached signature of all `/boot/kexec*.txt` file, verified by:
`if ! sha256sum `find $1/kexec*.txt` | gpgv $1/kexec.sig`
Which `gpgv` is basically just a script calling `gpg --verify $@`
Under /etc/functions's check_config function.

This detached signature file is created under kexec-sign-config:
```
param_files=`find $paramsdir/kexec*.txt`
if [ -z "$param_files" ]; then
        die "$paramsdir: No kexec parameter files to sign"
fi

for tries in 1 2 3; do
        if sha256sum $param_files | gpg \
                --digest-algo SHA256 \
                --detach-sign \
                -a \
                > $paramsdir/kexec.sig \
        ; then
                # successful - update the validated params
                check_config $paramsdir
                exit 0
        fi
done

```

So...
/boot/kexec_hashes.txt: is a sha256sum (digest) file of all /boot content
/boot/kexec.sig is a detached signature file of all `kexec*` files, including the kexec_hashes.txt file.

The function verify_global_hashes (which is duplicated in both gui-init and kexec-select-boot) verifies that the copied kexec_hashes.txt over /tmp matches the actual calculated hashes prior of booting.

On boot, `kexec.sign` is verifying detached signed integrity/authenticity of that signed file against the public key fused in rom, which was detached signed with the USB Security dongle's private key, protected in the USB Security dongle's smartcard, unlocked to sign with GPG User PIN.

Hope this clarifies things a bit

Reply:
I have reworded my paragraph accordingly and incorporated your descriptions into the section.  This is a useful description of the process.  It might be better in a different location in the docs but since this is a troubleshooting document I think it is fine to put it here for now.